### PR TITLE
Updated linter configs

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
Enabled and fixed these ruff plugins:
- `PLC`
- `A`

More to enable and fix soon.

### Issue link
#784 

---

### Code Checklist
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](/docs/source/templates/pull_request_template_review_code.md).

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](/docs/source/templates/pull_request_template_review_qa.md).

### CODE REVIEW

**Comment by dekkers** [comment]
What is the problem with shadowing builtins? The flake8-builtins example would be caught by both CI and code review. There are a lot of generic named python builtins and adding a trailing _ to them is just ugly, so we should only do that if there realy is a benefit, which I currently fail to see.

**Comment by ammar92** [comment]
> What is the problem with shadowing builtins? The flake8-builtins example would be caught by both CI and code review. There are a lot of generic named python builtins and adding a trailing _ to them is just ugly, so we should only do that if there realy is a benefit, which I currently fail to see.

I guess the real benefit is avoiding confusion and spreading consistency.

Shadowing builtins is even worse than shadowing outer scoped variables. It's confusing, linters and static analysis of proper IDEs and other languages nag about it for good reasons. There are about 158 builtins (Python 3.11), I'm sure there's a magnitude of other variable names one can think of. For example, some function arguments named `type` could've also been `object_type` or `plugin_type`. Also, it doesn't make sense, for example, to only add an underscore to that `type` argument if we wish to use `type` (as a function) in the function body, why not be consistent everywhere?

File: boefjes/boefjes/plugins/kat_adr_validator/main.py [hunk 1]
```diff
@@ -17,8 +17,8 @@ def run_adr_validator(url: str) -> str:
 
 
 def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
-    input = boefje_meta.arguments["input"]
-    api_url = input["api_url"]
+    input_ = boefje_meta.arguments["input"]
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_adr_validator/main.py:20]: I'd use "input_ooi"

File: boefjes/boefjes/plugins/kat_leakix/main.py [hunk 2]
```diff
@@ -21,13 +21,13 @@ def run(boefje_meta: BoefjeMeta) -> List[Tuple[set, Union[bytes, str]]]:
     else:
         raise NameError(f'Expected an IPAddress of Hostname, but got pk "{pk}"')
 
-    for type in ("leak", "service"):
+    for type_ in ("leak", "service"):
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_leakix/main.py:24]: "scope"

File: boefjes/boefjes/plugins/kat_nuclei/main.py [hunk 3]
```diff
@@ -7,15 +7,15 @@
 NUCLEI_IMAGE = "projectdiscovery/nuclei:latest"
 
 
-def verify_hostname_meta(input):
+def verify_hostname_meta(input_):
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_nuclei/main.py:10]: "input_ooi"

File: boefjes/boefjes/plugins/kat_nuclei_cve/main.py [hunk 4]
```diff
@@ -7,15 +7,15 @@
 NUCLEI_IMAGE = "projectdiscovery/nuclei:v2.9.1"
 
 
-def verify_hostname_meta(input):
+def verify_hostname_meta(input_):
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_nuclei_cve/main.py:10]: "input_ooi"

File: boefjes/boefjes/plugins/kat_nuclei_exposed_panels/main.py [hunk 5]
```diff
@@ -7,15 +7,15 @@
 NUCLEI_IMAGE = "projectdiscovery/nuclei:v2.9.1"
 
 
-def verify_hostname_meta(input):
+def verify_hostname_meta(input_):
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_nuclei_exposed_panels/main.py:10]: "input_ooi"

File: boefjes/boefjes/plugins/kat_nuclei_take_over/main.py [hunk 6]
```diff
@@ -7,15 +7,15 @@
 NUCLEI_IMAGE = "projectdiscovery/nuclei:v2.9.1"
 
 
-def verify_hostname_meta(input):
+def verify_hostname_meta(input_):
```
Comment (Donnype) [inline boefjes/boefjes/plugins/kat_nuclei_take_over/main.py:10]: "input_ooi"

File: keiko/keiko/keiko.py [hunk 7]
```diff
@@ -43,17 +43,17 @@
 )
 
 
-def latex_escape(input: Any) -> str:
+def latex_escape(input_: Any) -> str:
```
Comment (Donnype) [inline keiko/keiko/keiko.py:46]: "input_ooi"

File: mula/scheduler/repositories/sqlalchemy/task_store.py [hunk 8]
```diff
@@ -23,7 +23,7 @@ def __init__(self, datastore: SQLAlchemy) -> None:
     def get_tasks(
         self,
         scheduler_id: Optional[str],
-        type: Optional[str],
+        type_: Optional[str],
```
Comment (Donnype) [inline mula/scheduler/repositories/sqlalchemy/task_store.py:26]: "task_type"

File: mula/scheduler/queues/pq.py [hunk 9]
```diff
@@ -223,7 +223,7 @@ def is_item_on_queue(self, p_item: models.PrioritizedItem) -> bool:
 
         return True
 
-    def is_item_on_queue_by_hash(self, hash: str) -> bool:
+    def is_item_on_queue_by_hash(self, hash_: str) -> bool:
```
Comment (Donnype) [inline mula/scheduler/queues/pq.py:226]: "item_hash"

File: mula/scheduler/repositories/stores.py [hunk 10]
```diff
@@ -19,7 +19,7 @@ def __init__(self) -> None:
     def get_tasks(
         self,
         scheduler_id: Optional[str],
-        type: Optional[str],
+        type_: Optional[str],
```
Comment (Donnype) [inline mula/scheduler/repositories/stores.py:22]: "task_type"

File: mula/scheduler/server/server.py [hunk 11]
```diff
@@ -229,9 +229,9 @@ def patch_scheduler(self, scheduler_id: str, item: models.Scheduler) -> Any:
     def list_tasks(
         self,
         request: fastapi.Request,
-        scheduler_id: Optional[str] = None,
-        type: Optional[str] = None,
-        status: Optional[str] = None,
+        scheduler_id: Union[str, None] = None,
+        type_: Union[str, None] = None,
```
Comment (Donnype) [inline mula/scheduler/server/server.py:233]: "task_type"

File: mula/scheduler/server/server.py [hunk 12]
```diff
@@ -229,9 +229,9 @@ def patch_scheduler(self, scheduler_id: str, item: models.Scheduler) -> Any:
     def list_tasks(
         self,
         request: fastapi.Request,
-        scheduler_id: Optional[str] = None,
-        type: Optional[str] = None,
-        status: Optional[str] = None,
+        scheduler_id: Union[str, None] = None,
```
Comment (Donnype) [inline mula/scheduler/server/server.py:232]: Why replace `Optional[str]` with `Union[str, None]`?

File: mula/scheduler/server/server.py [hunk 13]
```diff
@@ -245,7 +245,7 @@ def list_tasks(
 
             results, count = self.ctx.task_store.get_tasks(
                 scheduler_id=scheduler_id,
-                type=type,
+                type_=type_,
```
Comment (Donnype) [inline mula/scheduler/server/server.py:248]: "task_type"

File: rocky/tools/forms/finding_type.py [hunk 14]
```diff
@@ -87,8 +87,8 @@ def clean_id(self):
 
         return data
 
-    def check_finding_type_existence(self, id):
-        _, created = OOIInformation.objects.get_or_create(id=f"KATFindingType|{id}")
+    def check_finding_type_existence(self, id_):
```
Comment (Donnype) [inline rocky/tools/forms/finding_type.py:90]: finding_type_id

File: rocky/rocky/scheduler.py [hunk 15]
```diff
@@ -178,7 +182,7 @@ def list_tasks(
     def get_lazy_task_list(
         self,
         scheduler_id: str,
-        type: Optional[str] = None,
+        type_: Optional[str] = None,
```
Comment (Donnype) [inline rocky/rocky/scheduler.py:185]: "task_type"

File: rocky/rocky/scheduler.py [hunk 16]
```diff
@@ -189,7 +193,7 @@ def get_lazy_task_list(
         return LazyTaskList(
             self,
             scheduler_id=scheduler_id,
-            type=type,
+            type=type_,
```
Comment (Donnype) [inline rocky/rocky/scheduler.py:196]: "task_type"

File: octopoes/bits/oois_in_headers/oois_in_headers.py [hunk 17]
```diff
@@ -38,15 +38,15 @@ def run(
     if input_ooi.key.lower() == "content-security-policy":
         urls_and_hostname = re.findall(r"(?:(?:https?|ftp):\/\/)?[\w/\-?=%.]+\.[\w/\-&?=%.]+", input_ooi.value)
 
-        for object in urls_and_hostname:
+        for object_ in urls_and_hostname:
```
Comment (Donnype) [inline octopoes/bits/oois_in_headers/oois_in_headers.py:41]: url_or_hostname

File: octopoes/bits/spf_discovery/internetnl_spf_parser.py [hunk 18]
```diff
@@ -131,9 +131,9 @@ def _check_domain_end(tokens):
     + Optional(dual_cidr_length)
 )
 include = Combine(Optional(qualifier) + CaselessLiteral("include:") + domain_spec)
-all = Combine(Optional(qualifier) + CaselessLiteral("all"))
+all_ = Combine(Optional(qualifier) + CaselessLiteral("all"))
```
Comment (Donnype) [inline octopoes/bits/spf_discovery/internetnl_spf_parser.py:134]: :thinking: "all_combined"?

File: mula/scheduler/server/server.py (hunk 12 above)
Comment (ammar92) [reply mula/scheduler/server/server.py:232]: Not sure, I got this after a merge

**Pull request merged**